src/sentry/grouping/variants.py

```diff
@@ -147,7 +148,10 @@ def expose_fingerprint_dict(values, info):
 
     matched_rule = info.get("matched_rule")
     if matched_rule:
-        rv["matched_rule"] = matched_rule["text"]
+        # TODO: Before late October 2024, we didn't store the rule text along with the matched rule,
+        # meaning there are still events out there whose `_fingerprint_info` entry doesn't have it.
+        # Once those events have aged out (in February or so), we can remove the default value here.
+        rv["matched_rule"] = matched_rule.get("text", FingerprintRule.from_json(matched_rule).text)
```

Comment:
thanks for fixing this! do we need a quick test?

Reply:
I debated that, but since it's a) just a temporary fix, and b) essentially just a reversion of that part of the PR, wasn't sure it was worth the time (and overhead of making sure it also gets deleted). WDYT?

Reply:
ah yeah, that makes sense

Reply:
LOL - in the meantime I decided I should stop being lazy and I added one.